src/store/reducers/filter.ts

```diff
@@ -52,6 +54,7 @@ const FilterReducer = (
       };
     case 'ADD_FILTER': {
       const isValid = isValidFilter(action.payload);
+      isValid && dispatch({ type: 'SET_PAGE', payload: 1 });
```

Comment:
you don't need to use dispatch. You can update the `page` value directly

Reply:
just by returning page value in the return statement?